libs/vertexai/langchain_google_vertexai/vision_models.py

```diff
@@ -0,0 +1,276 @@
+                - Local file path
+                - Remote url
+
+        Returns:
+            Captions generated from every prompt.
+        """
+
+        generations = [self._generate_one(prompt=prompt) for prompt in prompts]
+
+        return LLMResult(generations=generations)
+
+    def _generate_one(self, prompt: str) -> List[Generation]:
+        """Generates the captions for a single prompt.
+
+        Args:
+            prompt: Image url for the generation.
+
+        Returns:
+            List of generations
+        """
+        image_loader = ImageBytesLoader(project=self.project)
+        image_bytes = image_loader.load_bytes(prompt)
+        image = Image(image_bytes=image_bytes)
+        caption_list = self._get_captions(image=image)
+        return [Generation(text=caption) for caption in caption_list]
+
+
+class VertexAIImageCaptioningChat(_BaseVertexAIImageCaptioning, BaseChatModel):
+    """Implementation of the Image Captioning model as a chat."""
+
+    def _generate(
+        self,
+        messages: List[BaseMessage],
+        stop: List[str] | None = None,
+        run_manager: CallbackManagerForLLMRun | None = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        """Generates the results.
+
+        Args:
+            messages: List of messages. Currently only one message is supported.
+                The message must contain a string representation of the image.
+                Currently supported are:
+                - Google Cloud Storage URI
+                - B64 encoded string
+                - Local file path
+                - Remote url
+        """
+
+        if len(messages) != 1:
+            raise ValueError(
+                "Image captioning only works with one message: the image. "
+                f"instead got {len(messages)}"
+            )
+
+        message = messages[0]
+        image = self._get_image_from_message(message)
+        captions = self._get_captions(image)
+
+        generations = [
+            ChatGeneration(message=AIMessage(content=caption)) for caption in captions
+        ]
+
+        return ChatResult(generations=generations)
+
+
+class VertexAIVisualQnAChat(_BaseImageTextModel, BaseChatModel):
+    """Chat implementation of a visual QnA model"""
+
+    def _generate(
+        self,
+        messages: List[BaseMessage],
+        stop: List[str] | None = None,
+        run_manager: CallbackManagerForLLMRun | None = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        """Generates the results.
+
+        Args:
+            messages: List of messages. The first message should contain a
+            string representation of the image.
+                Currently supported are:
+                    - Google Cloud Storage URI
+                    - B64 encoded string
+                    - Local file path
+                    - Remote url
+                There has to be at least other message with the first question.
+        """
+
+        if len(messages) < 2:
```

Comment:
I don't think we should expect two messages, I'd rather suggest we should expect one with multiple content parts:
```
image_message = {
        "type": "image_url",
        "image_url": {"url": gcs_url},
}
text_message = {
        "type": "text",
        "text": "What is shown in this image?",
    }
message = HumanMessage(content=[text_message, image_message])
```

Reply:
This is in order for it to work with the chat interface, each turn creates one message right? Maybe I can do so that the first is multimodal?

Reply:
yes, let's make it multimodal and it should expect only one message, so invocation would look smth like this:

```
response = model.invoke(
        input=[
            HumanMessage(content=[text_message, image_message])
        ]
    )
```

Reply:
Changed!